engine/src/main/java/io/camunda/zeebe/engine/api/records/RecordBatch.java

```diff
@@ -99,6 +99,10 @@ public Spliterator<ImmutableRecordBatchEntry> spliterator() {
     return recordBatchEntries.spliterator();
   }
 
+  /**
+   * This exception is part of the contract with the engine. When a processor receives an exception
+   * it can check if it's an instance of this exception and handle it accordingly.
```

Comment:
>When a processor receives an exception it can check if it's an instance of this exception and handle it accordingly.

This part is specific to the engine, it might change (although it probably won't) 🤷 

🔧 Let's simplify this to something like: `The engine may handle this exception explicitly`